### PR TITLE
Fix: x-ms-paths merged into path in oai2-to-oai3 converter

### PR DESCRIPTION
**Pull request by timotheeguerin**
Issue is that `x-ms-paths` are merged into `paths` immediately when converting from swagger to openapi 3. This is incorrect as using query param in path is still not allowed in openapi 3.
Move the merge to be latter in the pipeline.

### CODE REVIEW

File: packages/libs/oai2-to-oai3/test/resources/conversion/oai3/ApiManagementClient-openapi.json [hunk 1]
```diff
@@ -5808,7 +5808,9 @@
         },
         "x-ms-requestBody-index": 4
       }
-    },
+    }
+  },
+  "x-ms-paths": {
     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/apis/{apiId}?export=true": {
```
Comment (pakrym) [inline packages/libs/oai2-to-oai3/test/resources/conversion/oai3/ApiManagementClient-openapi.json:5814]: worth adding a test that has a parameter defined in a query?

**Comment by pakrym** [comment]
Works

File: packages/libs/oai2-to-oai3/test/resources/conversion/oai3/ApiManagementClient-openapi.json (hunk 1 above)
Comment (timotheeguerin) [reply packages/libs/oai2-to-oai3/test/resources/conversion/oai3/ApiManagementClient-openapi.json:5814]: for what this test is doing it shouldn't make a different between the content of the path. The semantic validator was never supposed to validate x-ms-paths

Comment (pakrym) [reply packages/libs/oai2-to-oai3/test/resources/conversion/oai3/ApiManagementClient-openapi.json:5814]: Why not? SHould be fine for the path part.

Comment (timotheeguerin) [reply packages/libs/oai2-to-oai3/test/resources/conversion/oai3/ApiManagementClient-openapi.json:5814]: well when I added the path validator it was for the openapi paths specs. I could add another one for the x-ms-paths in particular but i'll have first to double check the actual spec for that and what we should actually be validating

**Pull request merged**